### PR TITLE
model: filter threshold parameters added

### PR DESCRIPTION
**Pull request by ovese**
Adding parameters to filter model in core.cpp. Threshold parameters added and undergoing improvements

### CODE REVIEW

File: lib/include/irritator/core.hpp [hunk 1]
```diff
@@ -6023,6 +6027,7 @@ max_size_in_bytes() noexcept
                sizeof(cross),
                sizeof(time_func),
                sizeof(accumulator_2),
+               sizeof(filter),
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:6030]: This change must be merge with the previous patch.

File: lib/include/irritator/core.hpp [hunk 2]
```diff
@@ -6269,6 +6274,8 @@ dynamics_typeof() noexcept
         return dynamics_type::time_func;
     if constexpr (std::is_same_v<Dynamics, accumulator_2>)
         return dynamics_type::accumulator_2;
+    if constexpr (std::is_same_v<Dynamics, filter>)
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:6277]: This change must be merge with the previous patch.

File: lib/include/irritator/core.hpp [hunk 3]
```diff
@@ -6448,6 +6455,8 @@ struct simulation
             return f(*reinterpret_cast<accumulator_2*>(&mdl.dyn), args...);
         case dynamics_type::time_func:
             return f(*reinterpret_cast<time_func*>(&mdl.dyn), args...);
+        case dynamics_type::filter:
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:6458]: This change must be merge with the previous patch.

File: lib/include/irritator/core.hpp [hunk 4]
```diff
@@ -6576,6 +6585,8 @@ struct simulation
                      args...);
         case dynamics_type::time_func:
             return f(*reinterpret_cast<const time_func*>(&mdl.dyn), args...);
+        case dynamics_type::filter:
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:6588]: This change must be merge with the previous patch.

File: lib/include/irritator/core.hpp [hunk 5]
```diff
@@ -6879,6 +6890,7 @@ struct simulation
         case dynamics_type::constant:
         case dynamics_type::cross:
         case dynamics_type::time_func:
+        case dynamics_type::filter:
```
Comment (quesnel) [inline lib/include/irritator/core.hpp:6893]: This change must be merge with the previous patch.

**Review by quesnel — CHANGES_REQUESTED** [review]
All changes in switch must be merge with the previous patch to fix all warnings in North gcc and msvc.

**Comment by ovese** [comment]
I have made the merger of the commits. could you let me know if its ok now

**Comment by quesnel** [comment]
I am merging the patch myself. I delete the unnecessary parts. Thank you.

**Pull request closed without merging**